### ISSUE TITLE
I2S playback fails on M5 Atom

### ISSUE THREAD

**Issue by awillecke** [Romkabouter/ESP32-Rhasspy-Satellite#101]
Hi, first, thanks for this cool project!

I tried to compile using `device_type = 0` and flash it on an M5 Atom board with `pio run --target upload`, which went well at first.
However, there seems to be a problem with playback. 

When I sent TTS audio via Rhasspy, the following errors appear in the logs and no audio is played:

```
....
Send TtsEvent from sayTopic
Enter Tts
Samplerate: 16000, Channels: 1, Format: 1, Bits per Sample: 16, Start: 44
Send PlayBytesEvent
Send BeginPlayAudioEvent in PlayBytesEvent
BeginPlayAudioEvent in Tts
Enter TtsPlay
E (2061974) I2S: i2s_check_set_mclk(251): ESP32 only support to set GPIO0/GPIO1/GPIO3 as mclk signal, error GPIO number:-1287651329
E (2061977) I2S: i2s_set_pin(312): mclk config failed
Done
Send StreamAudioEvent
Send EndPlayAudioEvent in StreamAudioEvent
EndPlayAudioEvent in TtsPlay
Enter Tts
E (2063243) I2S: i2s_check_set_mclk(251): ESP32 only support to set GPIO0/GPIO1/GPIO3 as mclk signal, error GPIO number:-1287651329
E (2063251) I2S: i2s_set_pin(312): mclk config failed
Send IdleEvent from sayFinishedTopic
IdleEvent in Tts
Enter Idle
....
```

Do you have an idea what could cause this issue?

**Comment by erte1** [comment]
Not sure of your issue, but would you mind sharing your settings.ini general info as well as how you have your rhasspy base setup?  I'm not able to get the satellite and base to interact in any meaningful way.

**Comment by Romkabouter** [comment]
Your settings are ok, but the error message is this "ESP32 only support to set GPIO0/GPIO1/GPIO3 as mclk signal"
That is the reason you cannot hear a sound, it might indeed be a revision issue on the M5.
Since I have no such device, the only thing I can do is google on the error and maybe you can do as well and try to fix the issue

**Comment by awillecke** [comment]
Hey, I just found something else that might be related. Here is the log after resetting the device:

```
M5Atom initializing...OK
Loading configuration
{
  "siteid": "default",
  "mqtt_host": ***,
  "mqtt_port": 1883,
  "mqtt_user": ***,
  "mqtt_pass": ***,
  "mute_input": false,
  "mute_output": false,
  "amp_output": 0,
  "brightness": 15,
  "hotword_brightness": 15,
  "hotword_detection": 1,
  "volume": 90,
  "gain": 7,
  "animation": 0
}
Creating I2Stask
Enter WifiDisconnected
Enter WifiConnected
Connected to Wifi with IP: ***, SSID: ***, BSSID ***, RSSI: -51
Enter MQTTDisconnected
Connecting MQTT: ***, 1883
Enter MQTTConnected
Connected as default
Connected to asynch MQTT!
Enter Idle
E (659) I2S: i2s_driver_uninstall(1957): I2S port 0 has not installed
E (662) I2S: i2s_check_set_mclk(251): ESP32 only support to set GPIO0/GPIO1/GPIO3 as mclk signal, error GPIO number:-1287651329
E (666) I2S: i2s_set_pin(312): mclk config failed
```

So, something went wrong even before the "ESP32 only support to set GPIO0/GPIO1/GPIO3 as mclk signal" error, issued by playing TTS.

**Comment by awillecke** [comment]
Seems like the MCK pin was uninitialized. With this fix, I am now able to talk with the device :)

**Comment by Romkabouter** [comment]
Good find, I will test it with my M5 as well to see if that still works

**Comment by erte1** [comment]
Can confirm - the change works on this end when after making the proposed update

**Comment by Romkabouter** [comment]
Fixed with #103
